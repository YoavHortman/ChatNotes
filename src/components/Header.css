.Header__root {
    font-weight: bold;
    background: #d3d3d361;
    border-radius: 3px;
    padding: 10px;
    margin-bottom: 10px;
}