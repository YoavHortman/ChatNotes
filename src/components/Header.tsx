import * as React from "react";

interface HeaderProps {
    noteCount: number;
}

interface HeaderState {
}

export class Header extends React.Component<HeaderProps, HeaderState> {
    constructor(props: HeaderProps) {
        super(props);
    }

    render() {
        return (
            <div className={"Header__root"}>
                {this.props.noteCount} {this.props.noteCount === 1 ? "item" : "items"}
            </div>
        );
    }
}
