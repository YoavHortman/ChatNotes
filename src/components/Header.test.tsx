import * as React from "react";
import * as renderer from "react-test-renderer";
import {Header} from "./Header";

it('Should render correctly with the text "1 item"', () => {
    const tree = renderer
        .create(<Header noteCount={1}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Should render correctly with the text "2 items"', () => {
    const tree = renderer
        .create(<Header noteCount={2}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});