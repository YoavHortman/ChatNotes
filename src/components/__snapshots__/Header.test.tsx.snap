// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly with the text "1 item" 1`] = `
<div
  className="Header__root"
>
  1
   
  item
</div>
`;

exports[`Should render correctly with the text "2 items" 1`] = `
<div
  className="Header__root"
>
  2
   
  items
</div>
`;
